src/megolm/inbound_group_session.rs

```diff
@@ -344,64 +329,36 @@ impl InboundGroupSession {
 
 #[derive(Serialize, Deserialize)]
 pub struct InboundGroupSessionPickle {
-    initial_ratchet: RatchetPickle,
-    latest_ratchet: RatchetPickle,
+    initial_ratchet: Ratchet,
     signing_key: Ed25519PublicKey,
     #[allow(dead_code)]
     signing_key_verified: bool,
 }
 
 impl InboundGroupSessionPickle {
-    pub fn unpickle(self) -> Result<InboundGroupSession, InboundGroupSessionUnpicklingError> {
-        self.try_into()
+    pub fn encrypt(self, pickle_key: &[u8; 32]) -> String {
+        pickle(&self, pickle_key)
     }
-}
 
-impl TryFrom<InboundGroupSessionPickle> for InboundGroupSession {
-    type Error = InboundGroupSessionUnpicklingError;
-
-    fn try_from(pickle: InboundGroupSessionPickle) -> Result<Self, Self::Error> {
-        Ok(Self {
-            initial_ratchet: pickle
-                .initial_ratchet
-                .try_into()
-                .map_err(InboundGroupSessionUnpicklingError::InvalidInitialRatchet)?,
-            latest_ratchet: pickle
-                .latest_ratchet
-                .try_into()
-                .map_err(InboundGroupSessionUnpicklingError::InvalidLatestRatchet)?,
-            signing_key: pickle.signing_key,
-            signing_key_verified: pickle.signing_key_verified,
-        })
+    pub fn from_encrypted(ciphertext: &str, pickle_key: &[u8; 32]) -> Result<Self, PickleError> {
+        unpickle(ciphertext, pickle_key)
```

Comment:
The meaning of the method should be pretty clear, but let's still add a doc comment for the method.